patches/minecraft/net/minecraft/data/AdvancementProvider.java.patch

```diff
@@ -0,0 +1,38 @@
+--- a/net/minecraft/data/AdvancementProvider.java
++++ b/net/minecraft/data/AdvancementProvider.java
+@@ -24,11 +24,15 @@
+    private static final Gson field_204024_b = (new GsonBuilder()).setPrettyPrinting().create();
+    private final DataGenerator field_204025_c;
+    private final List<Consumer<Consumer<Advancement>>> field_204283_d = ImmutableList.of(new EndAdvancements(), new HusbandryAdvancements(), new AdventureAdvancements(), new NetherAdvancements(), new StoryAdvancements());
++   private net.minecraftforge.common.data.ExistingFileHelper fileHelper;
+ 
++   @Deprecated
+    public AdvancementProvider(DataGenerator p_i48869_1_) {
+       this.field_204025_c = p_i48869_1_;
+    }
+ 
++   public AdvancementProvider(DataGenerator generatorIn, net.minecraftforge.common.data.ExistingFileHelper fileHelperIn){ this.field_204025_c = generatorIn; this.fileHelper = fileHelperIn;}
++
+    public void func_200398_a(DirectoryCache p_200398_1_) throws IOException {
+       Path path = this.field_204025_c.func_200391_b();
+       Set<ResourceLocation> set = Sets.newHashSet();
+@@ -47,12 +51,19 @@
+          }
+       };
+ 
++      registerAdvancement(consumer, fileHelper);
++
++      if(this.getClass() == AdvancementProvider.class)//Forge: Subclasses don't need this.
```

Comment:
Are you 100% certain that this won't ever become necessary for other mods?

Reply:
yes since in the loop only vanilla Advancements will be generated and mods should use `registerAdvancement` which in essence does the same as the loop.

Reply:
This should be evident from the code, not from looking at this conversation :)
", use registerAdvancement instead" would suffice